python/social_bot/envs/gazebo_base.py

```diff
@@ -82,6 +84,39 @@ def render(self, mode='human'):
         Args:
             mode (str): 'human' and 'rgb_array' is supported.
         """
+        if self._rendering_camera is None:
+            render_camera_sdf = """
+            <?xml version='1.0'?>
+            <sdf version ='1.6'>
+            <model name ='render_camera'>
+                <static>1</static>
+                <pose>%s</pose>
+                <link name="link">
+                    <sensor name="camera" type="camera">
+                        <camera>
+                        <horizontal_fov>0.95</horizontal_fov>
+                        <image>
+                            <width>640</width>
+                            <height>480</height>
+                        </image>
+                        <clip>
+                            <near>0.1</near>
+                            <far>100</far>
+                        </clip>
+                        </camera>
+                        <always_on>1</always_on>
+                        <update_rate>30</update_rate>
+                        <visualize>true</visualize>
+                    </sensor>
+                </link>
+            </model>
+            </sdf>
+            """
+            render_camera_sdf = render_camera_sdf % self._rendering_cam_pose
+            self._world.insertModelFromSdfString(render_camera_sdf)
+            time.sleep(0.2)
+            self._world.step(20)
+            self._rendering_camera = self._world.get_agent('render_camera')
         if mode == 'human':
```

Comment:
Seems that human rendering is still gazebo. Why moving the inserting process above could making human rendering consistent with rgb_array mode?

Reply:
somehow world.step + adding the camera causes random generator to be in a different state for the human rendering case vs rgb_array, even with the same random seed.

Now with this change and some other change in alf, pseudo randomness is exactly the same for the two modes.